packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -215,9 +215,10 @@ SurfaceTextureEntry getFlutterTexture() {
   public void takePicture(String filePath, @NonNull final Result result) {
     final File file = new File(filePath);
 
-    if (file.exists()) {
-      result.error(
-          "fileExists", "File at path '" + filePath + "' already exists. Cannot overwrite.", null);
+    FilePathValidator validator = new FilePathValidator(filePath);
+    validator.validate();
+    if (!validator.isValid()) {
+      result.error("filePathInvalid", validator.getErrorMessage(), null);
```

Comment:
Changing the error code is a breaking change, I suggest we keep the error code to be `fileExists` to avoid a major version bump.